noir-projects/noir-contracts/contracts/app_subscription_contract/src/main.nr

```diff
@@ -11,20 +11,20 @@ contract AppSubscriptionContract {
     use dep::aztec::{
         context::{PrivateContext, Context}, note::{note_header::NoteHeader, utils as note_utils},
         oracle::get_public_key::get_public_key,
-        state_vars::{map::Map, singleton::Singleton, public_state::PublicState, stable_public_state::StablePublicState}
+        state_vars::{Map, PrivateMutable, PublicMutable, SharedImmutable}
     };
     use dep::authwit::{account::AccountActions, auth_witness::get_auth_witness, auth::assert_current_call_valid_authwit};
 
     use crate::subscription_note::{SubscriptionNote, SUBSCRIPTION_NOTE_LEN};
 
     struct Storage {
-        // todo change this to immutable singletons because it's only needed in private
-        target_address: StablePublicState<AztecAddress>,
-        subscription_token_address: StablePublicState<AztecAddress>,
-        subscription_recipient_address: StablePublicState<AztecAddress>,
-        subscription_price: StablePublicState<Field>,
-        subscriptions: Map<AztecAddress, Singleton<SubscriptionNote>>,
-        gas_token_address: StablePublicState<AztecAddress>,
+        // todo change this to PrivateImmutable because it's only needed in private
+        target_address: SharedImmutable<AztecAddress>,
```

Comment:
It might only be needed in private. But any user needed it will then need to insert it manually into their PXE. It seems like the perfect example of when you would be using shared.